snap-app/src/solver/solver.js

```diff
@@ -1,23 +1,43 @@
 import React from "react";
 import { postJson } from "../fetch";
 import "./solver.css";
+import { withRouter } from "react-router-dom"
 
-export default class Solver extends React.Component {
+class Solver extends React.Component {
 
     constructor(props) {
         super(props);
         this.state = {
             query: '',
             wordLengths: '',
             canRearrange: false,
-
             loading: false,
             results: [],
         };
     }
 
     componentDidMount() {
         window.addEventListener("keyup", this.handleKey);
+        const searchParams = new URLSearchParams(this.props.location.search);
```

Comment:
Could also do JSON.parse and JSON.stringify on the entire state object to avoid much of this parsing logic - I did something like that in [wordsearch.js](https://github.com/kevinychen/snap2/blob/master/snap-app/src/wordsearch/wordsearch.js).